PyRoute/TradeCodes.py

```diff
@@ -80,6 +96,148 @@ def __init__(self, initial_codes):
         self.dcode = list(self.dcode) + self.colony + self.owner
         self.dcode = sorted(self.dcode)
 
+        self.trim_ill_formed_residual_codes()
+
+    def _process_sophonts_and_homeworlds(self, initial_codes):
+        self.homeworld_list = []
+        self.sophont_list = []
+        homeworlds_found = []
+        self.sophont_list = [code for code in self.codes if
```

Comment:
Same note here about creating using `sophont_search = re.compile()` and `sophont_search.match()`

Reply:
Compiled and pushed out